rgw/v2/tests/s3_swift/test_s3cmd.py

```diff
@@ -0,0 +1,160 @@
+"""
+test_s3cmd - Test s3cmd operation on cluster
+
+Usage: test_s3cmd.py -c <input_yaml>
+
+<input_yaml>
+    Note: Following yaml can be used
+    test_s3cmd.yaml
+
+Operation:
+    Create an user
+    Create a bucket with user credentials
+    Upload a file to bucket
+    Delete uploaded object
+    Delete bucket
+"""
+
+
+import argparse
+import json
+import logging
+import os
+import requests
+import sys
+import time
+import traceback
+import uuid
+import yaml
+sys.path.append(os.path.abspath(os.path.join(__file__, "../../../..")))
+
+
+from v2.lib.admin import UserMgmt
+from v2.lib.exceptions import TestExecError, RGWBaseException
+from v2.lib import manage_data
+from v2.lib import resource_op as swiftlib
+from v2.lib.rgw_config_opts import CephConfOp, ConfigOpts
+from v2.lib.swift.auth import Auth
+from v2.lib.s3.s3cmd import auth as s3_auth
+from v2.lib.s3.s3cmd.resource_op import S3CMD
+from v2.lib.s3.write_io_info import IOInfoInitialize, BasicIOInfoStructure
+from v2.utils import utils
+from v2.utils.log import configure_logging
+from v2.utils.utils import HttpResponseParser, RGWService
+from v2.utils.test_desc import AddTestInfo
+from v2.tests.s3_swift import reusable
+from v2.tests.s3_swift.test_swift_basic_ops import fill_container
+
+log = logging.getLogger()
+
+
+def test_exec(config):
+    """
+    Executes test based on configuration passed
+    Args:
+        config(object): Test configuration
+    """
+    io_info_initialize = IOInfoInitialize()
+    basic_io_structure = BasicIOInfoStructure()
+    io_info_initialize.initialize(basic_io_structure.initial())
+    umgmt = UserMgmt()
+    ceph_conf = CephConfOp()
+    rgw_service = RGWService()
+    # preparing data
+    user_name = str(uuid.uuid1()).split('-')[0]
+    tenant = 'tenant'
+    tenant_user_info = umgmt.create_tenant_user(tenant_name=tenant,
+                                                user_id=user_name,
+                                                displayname=user_name)
+    user_info = umgmt.create_subuser(tenant_name=tenant, user_id=user_name)
+
+    auth = Auth(user_info)
+    rgw = auth.do_auth()
+
+    ip_and_port = rgw.authurl.split('/')[2]
+    s3_auth.do_auth(tenant_user_info, ip_and_port)
+
+    bucket_name = utils.gen_bucket_name_from_userid(user_name, rand_no=0)
+
+    #Create a bucket
+    create_bucket = S3CMD(operation="mb")
+    create_bucket.command(params=["s3://{}".format(bucket_name)])
+    create_bucket_response = str(create_bucket.execute())
+    expected_response = "Bucket 's3://{}/' created".format(bucket_name)
+    error_message = 'Expected: %s, Actual: %s' % (
+        expected_response, create_bucket_response)
+    assert expected_response in create_bucket_response, error_message 
```

Comment:
dont use assert here, raise another exception `s3cmdCommandFailed`, create this exception class in exceptions.py

Reply:
Added exception for command execution step. Kept assertion as it is to verify command response. If response is not as expected, it will raise AssertionError.